src/transforms/transform-deployment.ts

```diff
@@ -182,7 +205,13 @@ export const transformDeployment = async (githubClient: GitHubAPI, newGitHubClie
 			logger
 		);
 
+		const allCommitsMessages = commitSummaries ? await extractMessagesFromCommitSummaries(commitSummaries) : "";
```

Comment:
I'd rather the function handle undefined `commitSummaries` than have it outside of it.  I'm not a fan of the 2 different types being returned.  one is a `CommitSummary[]` and the other is a string.  I think the `undefined` route should return either `undefined` or `[]`.  I tend to lean towards the latter as there's less possibilities of runtime errors being thrown.